ospd/protocol.py

```diff
@@ -146,6 +146,32 @@ def process_credentials_elements(cred_tree: Element) -> Dict:
 
         return credentials
 
+    @staticmethod
+    def process_alive_test_methods(cred_tree: Element, options: Dict) -> None:
+        """ Receive an XML object with the alive test methods to run
+        a scan with. Methods are added to the options Dict.
+
+        @param
+        <alive_test_methods>
+            </icmp></icmp>
+            </tcp_ack></tcp_ack>
+            </tcp_syn></tcp_syn>
+            </arp></arp>
+            </consider_alive>0</consider_alive>
+        </alive_test_methods>
+        """
+        for child in cred_tree:
+            if child.tag == 'icmp':
```

Comment:
Are these methods optional? shouldn't be added only if they are set? Not sure if it make sense to add and empty element to the options dict.

Reply:
All the options (imcp, tcp_ack ...) are optional. For example if you do not want icmp then you can either leave it out completely, set it to '0' or leave it empty. Everything not '1' is treated as not wanted.

Should I check if child.text is None and not include it in the options dict?

Just an additional remark. If <alive_test> and <alive_test_methods> is provided, <alive_test> takes precedence over <alive_test_methods>.

Reply:
I have now added a test for None. So if there is no text it will not be included options dict.
I also changed the argument description in the function.